src/lib.mo

```diff
@@ -344,14 +344,19 @@ module {
           return #err(err);
         }
       );
-      let result = await* accountManager.withdraw(to, amount);
+      let result = await* accountManager.withdraw(?p, to, amount, expectedFee);
       if (Result.isErr(result)) {
         // re-issue credit if unsuccessful
         creditRegistry.issue(#user p, amount);
       };
       result;
     };
 
+    /// Clears the credit registry of entries with values below the provided minimum.
+    /// The method is designed to prevent spam attacks.
+    /// Returns the number of entries cleared.
+    public func clean(min : Nat) : Int = creditRegistry.clean(min);
```

Comment:
Is it clear that this is save to do? Without looking at locks I mean.

Reply:
In general, considering that we do pre-issue/pre-burn in notifications and withdrawals, this is not particularly safe in theory. However, our locks are in the depositRegistry. Wouldn't it be better to create a separate data structure with locks? Another option, probably more reliable and safer, is to make sure everything is paused before cleaning.

Reply:
Let's remove `clean` functionality from this PR and introduce it later. First release of TokenHandler does not need it. Then we have more time to think about its safety.